File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int, int> m1;
        // set<int> s;
        vector<int> v1, v2;
        for(auto x:matches){
            m1[x[0]];   //Just note the element in map
            m1[x[1]]++;
            // s.insert(x[0]);
        }

        for(auto x:m1){
            if(x.second==0) v1.push_back(x.first);
            else if(x.second == 1) v2.push_back(x.first);
        }
        
        // for(auto x:s){
        //     if(m1.find(x) == m1.end()) v1.push_back(x);
        // }
        return {v1, v2};
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches):
        m1 = {}
        s = set()
        v1, v2 = [], []

        for x in matches:
            m1[x[1]] = m1.get(x[1], 0) + 1
            s.add(x[0])

        for key, value in m1.items():
            if value == 1:
                v2.append(key)

        for x in s:
            if x not in m1:
                v1.append(x)
                
        v1.sort()
        v2.sort()
        return [v1, v2]
